nifi-nar-bundles/nifi-standard-bundle/nifi-standard-parameter-providers/src/main/java/org/apache/nifi/parameter/OnePasswordParameterProvider.java

```diff
@@ -0,0 +1,216 @@
+                                                            .uri(URI.create(connectServer + GET_VAULTS))
+                                                            .header(AUTHORIZATION_HEADER_NAME, AUTHORIZATION_HEADER_VALUE + accessToken)
+                                                            .header(CONTENT_TYPE, APPLICATION_JSON)
+                                                            .retrieve();
+
+            final JsonNode vaultList = OBJECT_MAPPER.readTree(getVaultList.body());
+            final Iterator<JsonNode> vaultIterator = vaultList.elements();
+
+            // we iterate though each vault
+            while(vaultIterator.hasNext()) {
+                final JsonNode vault = vaultIterator.next();
+                final String vaultName = vault.get("name").asText();
+                final String vaultID = vault.get("id").asText();
+
+                final List<Parameter> parameters = new ArrayList<Parameter>();
+
+                final HttpResponseEntity getVaultItems = webClientServiceProvider.getWebClientService()
+                                                                .get()
+                                                                .uri(URI.create(connectServer + GET_VAULTS + "/" + vaultID + "/items"))
+                                                                .header(AUTHORIZATION_HEADER_NAME, AUTHORIZATION_HEADER_VALUE + accessToken)
+                                                                .header(CONTENT_TYPE, APPLICATION_JSON)
+                                                                .retrieve();
+
+                final JsonNode itemList = OBJECT_MAPPER.readTree(getVaultItems.body());
+                final Iterator<JsonNode> itemIterator = itemList.elements();
+
+                // we iterate though the items
+                while(itemIterator.hasNext()) {
+                    final JsonNode item = itemIterator.next();
+                    final String itemID = item.get("id").asText();
+
+                    final HttpResponseEntity getItemDetails = webClientServiceProvider.getWebClientService()
+                                                                    .get()
+                                                                    .uri(URI.create(connectServer + GET_VAULTS + "/" + vaultID + "/items/" + itemID))
+                                                                    .header(AUTHORIZATION_HEADER_NAME, AUTHORIZATION_HEADER_VALUE + accessToken)
+                                                                    .header(CONTENT_TYPE, APPLICATION_JSON)
+                                                                    .retrieve();
+
+                    final JsonNode itemDetails = OBJECT_MAPPER.readTree(getItemDetails.body());
+                    final String itemName = itemDetails.get("title").asText();
+                    final Iterator<JsonNode> itemFields = itemDetails.get("fields").elements();
```

Comment:
It seems like it would be helpful to divide this method into smaller methods for getting items versus getting vaults.